iothub/device/tests/DeviceClientTests.cs

```diff
@@ -1684,6 +1685,94 @@ public void DeviceClient_CreateFromAuthenticationMethod_SasTokenTimeToLiveRenewa
             authMethod.RefreshesOn.Should().BeCloseTo(expectedRefreshTimeFromSdkDefault, (int)buffer.TotalMilliseconds);
         }
 
+        [TestMethod]
```

Comment:
Unit tests to validate it throws when it should, and doesn't throw when it shouldn't. Any gaps?